### PR TITLE
Feat/Add login page

### PR DESCRIPTION
**Pull request by qwoprocks**
Summary:
- Add login, signup, confirm, logout, and delete
- Add confirmation page for the signup
- Update outdated npm packages
- Fix linting errors (Vetur + Typescript lint, refer to my comments in the thread below)
- Add profile page to view email + username + delete account

To fix:
- [ ] Currently, not enough visual / UX feedback to the user, might need to add more alerts and toasts
- [ ] Error handling might need to be managed better
- [ ] Find a way to tell the user that email already exists during sign up
- [ ] Find a way to redirect the user to confirmation page if they have not confirmed their email
- [ ] Add forget password?

### CODE REVIEW

File: ionic_user_interface/src/views/Login.vue [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    const onSubmit = (event: Event): boolean => {
+      event.preventDefault();
+      // Check credentials
+      if (!isValidEmail(emailText.value)) {
+        errorMsg.value = 'Invalid email.';
+        showErrorMsg.value = true;
+        return false;
+      }
+      if (!isValidPassword(passwordText.value)) {
+        errorMsg.value = 'Password has to be at least 6 characters.';
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/views/Login.vue:135]: Password minimum 8 characters

File: ionic_user_interface/src/views/Login.vue [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    IonLabel,
+    IonPage,
+    IonRow,
+    Header,
+  },
+  setup() {
+    // Email validation regex taken from https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript
+    const emailPattern =
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    const isLoggedIn: Ref<boolean> | undefined = inject('isLoggedIn');
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/views/Login.vue:105]: Maybe we can store the accessToken and just check if its an empty string or not

File: ionic_user_interface/src/views/Signup.vue [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+    const isValidEmail = (email: string): boolean => {
+      return emailPattern.test(email.toLowerCase());
+    };
+
+    const isValidPassword = (password: string): boolean => {
+      return password.length >= 6;
+    };
+
+    const isValidUsername = (username: string): boolean => {
```
Comment (yarkhinephyo) [inline ionic_user_interface/src/views/Signup.vue:145]: Lets limit to alphanumeric and space so won't be too funky

https://github.com/yarkhinephyo/yolo_bouldering/blob/f2f0b9f5b1afc41fcbb8a875c8b0f5522282eff8/lambda_backend/user_microservice/src/common/schema.ts#L1

**Comment by yarkhinephyo** [comment]
Added endpoint for a user to delete himself. Refer to the google docs API documentation

File: ionic_user_interface/tsconfig.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
     "allowSyntheticDefaultImports": true,
     "sourceMap": true,
     "baseUrl": ".",
-    "types": ["webpack-env", "jest"],
+    "types": ["webpack-env", "jest", "node"],
```
Comment (qwoprocks) [inline ionic_user_interface/tsconfig.json:14]: Resolves the error: `process` variable not found

File: vetur.config.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// vetur.config.js
```
Comment (qwoprocks) [inline vetur.config.js:1]: This file resolves the `@` webpack alias errors for the VSCode Vetur extension

**Comment by yarkhinephyo** [comment]
After logging out, can put an alert. But your choice ah.

---
Response when the user signed up but never confirm:
login -> HTTP 400
```
{
    "Message": "UserNotConfirmedException"
}
```
---
Response when the user's email is taken:
signup -> HTTP 400
```
{
    "Message": "UsernameExistsException"
}
```
---
Forget password we can make a new issue. The flow will probably be 1) get code in email 2) use the code and the new password together to change password

**Review by yarkhinephyo — APPROVED** [review]

**Pull request merged**